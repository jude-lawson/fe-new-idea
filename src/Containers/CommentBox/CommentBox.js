import React, { Component } from 'react';
import { CommentCard } from '../CommentCard/CommentCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class CommentBox extends Component {

  displayComments = () => {
    if (this.props.comments) {
      const displayedComments = this.props.comments.map((comment, index) => {
        return <CommentCard {...comment} key={index} />;
      });
      return displayedComments;
    }
  }

  render() {
    return (
      <main>
        {this.displayComments()}
      </main>
    );
  }
}

CommentBox.propTypes = {
  comments: PropTypes.array
};

export const mapStateToProps = (state) => ({
  comments: state.comments
});

export default connect(mapStateToProps)(CommentBox);