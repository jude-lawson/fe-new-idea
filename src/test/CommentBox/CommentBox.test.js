import { CommentBox, mapStateToProps } from '../../Containers/CommentBox/CommentBox';
import { shallow } from 'enzyme';
import React from 'react';

describe('CommentBox', () => {
  const mockState = [
    {body: 'I do or do not think that is a great idea', id: 11}
  ]

  it('should match snapshot', () => {
    let wrapper = shallow(<CommentBox comments={mockState} />);

    expect(wrapper).toMatchSnapshot();
  })
});

describe('mapStateToProps', () => {
  const mockState = {
    comments: [
      {body: 'I do or do not think that is a great idea', id: 11}
    ]
  };

  it('should return a props object with the correct keys', () => {
    const expected = {
      comments: [
        {body: 'I do or do not think that is a great idea', id: 11}
      ]
    }
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  })
})