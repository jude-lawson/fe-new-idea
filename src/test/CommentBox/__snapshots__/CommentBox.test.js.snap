// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentBox
    comments={
        Array [
            Object {
              "body": "I do or do not think that is a great idea",
              "id": 11,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          body="I do or do not think that is a great idea"
          id={11}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "body": "I do or do not think that is a great idea",
          "id": 11,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            body="I do or do not think that is a great idea"
            id={11}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "body": "I do or do not think that is a great idea",
            "id": 11,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
